File: ContentServer/content_server_10_5/service/otag.feeds.service/src/main/java/com/opentext/otag/cs/feeds/FeedItem.java
package com.opentext.otag.cs.feeds;

import org.codehaus.jackson.JsonNode;


public class FeedItem {

	public enum Provider{PulseStatus, PulseContent, PulseMentions, PulseFollows}

    public String username;
	public String userID;
	public String firstName;
	public String lastName;
	public String userTitle;
	public String userLocation;
	public String id;
	public String name = null;
	public Integer subType = null;
	public String header;
	public String body;
	public String footer;
	public Integer commentCount;
	public Integer likeCount;
	public Boolean isLiked;
	public Provider feed;
    public String otagID;
	public String createDate;
	public Integer seqNo;
	public Integer conversationID;	
	public boolean isFollowing;
	public String userPhotoSuffix;
	public String attachmentMimeTypeImgURL = null;
	public String attachmentID = null;
	public String attachmentName = null;
	public Integer attachmentSubType = null;
	public Integer inReplyToSeqNo = null;
	public String inReplyToUserID = null;
	public String inReplyToUserName = null;
	public String inReplyToFirstName = null;
	public String inReplyToLastName = null;
	
	public FeedItem(){}
	
	public FeedItem(JsonNode node){
		feed = Provider.valueOf(node.get("feed").asText());
		username = node.get("userName").getTextValue();
		userID = node.get("userID").asText();
		firstName = node.get("firstName").getTextValue();
		lastName = node.get("lastName").getTextValue();
		userTitle = node.get("userTitle").getTextValue();
		userLocation = node.get("userLocation").getTextValue();
		id = node.get("id").asText();
		if (feed == Provider.PulseContent){
			name = node.get("name").asText();
			subType = node.get("subType").asInt();
		}
		header = null;
		body = node.get("body").getTextValue();
		footer = null;
		commentCount = node.get("commentCount").asInt();
		likeCount = node.get("likeCount").asInt();
		isLiked = node.get("isLiked").asBoolean();
		otagID = null;
		createDate = node.get("createDate").getTextValue();
		seqNo = node.get("seqNo").asInt();
		conversationID = node.get("conversationID").asInt();
		isFollowing = node.get("isFollowing").asBoolean();
		JsonNode userPhotoSuffixNode = node.get("userPhotoSuffix");
		userPhotoSuffix = userPhotoSuffixNode.isNull() ? null : userPhotoSuffixNode.asText();
		if (node.has("attachmentID") && !node.get("attachmentID").isNull()){
			attachmentID = node.get("attachmentID").asText();
			attachmentMimeTypeImgURL = node.get("attachmentMimeTypeImgURL").asText();
			attachmentName = node.get("attachmentName").getTextValue();
			attachmentSubType = node.get("attachmentSubType").asInt();
		}
		if (node.has("inReplyTo") && !node.get("inReplyTo").isNull()){
			inReplyToSeqNo = node.get("inReplyTo").asInt();
			inReplyToUserID = node.get("inReplyToUserID").asText();
			inReplyToUserName = node.get("inReplyToUserName").getTextValue();
			inReplyToFirstName = node.get("inReplyToUserFirstName").getTextValue();
			inReplyToLastName = node.get("inReplyToUserLastName").getTextValue();
		}
	}
}

File: otsync.html5/otsync-apps-parent/src/app/www/modules/feed/model/feedItem.js
angular.module('FeedItem', ['userDisplayService'])
    .factory('FeedItem', ['$sessionService', '$displayMessageService', '$userDisplayService', function ($sessionService, $displayMessageService, $userDisplayService) {

        var FeedItem = function(feedItemData) {
			var _feedItemData = feedItemData;
			var _attachmentMimeTypeImgURL = $sessionService.getGatewayURL() + _feedItemData.attachmentMimeTypeImgURL;
			var _createDate = _feedItemData.createDate;
			var _feedItemType = _feedItemData.feed;
			var _firstName = _feedItemData.firstName;
			var _iconURL = $sessionService.getGatewayURL() + '/content/v5/users/' + _feedItemData.userID + '/photo';
			var _inReplyToSeqNum = _feedItemData.inReplyToSeqNo;
			var _lastName = _feedItemData.lastName;
			var _name = _feedItemData.name;
            var _username = _feedItemData.username || _feedItemData.displayName;

			this.getAttachmentID = function(){
				return _feedItemData.attachmentID;
			};

			this.getAttachmentName = function(){
				return _feedItemData.attachmentName;
			};

			this.getAttachmentMimeTypeImgURL = function(){
				return _attachmentMimeTypeImgURL;
			};

			this.getAttachmentSubtype = function(){
				return _feedItemData.attachmentSubType;
			};

			this.getCommentCount = function(){
				return _feedItemData.commentCount;
			};

			this.getCommentedOn = function(){
				var commentedOn = '';

				if (_feedItemType == "PulseContent" && _inReplyToSeqNum != null){
					commentedOn = $displayMessageService.translate("COMMENTED ON") + " " + _name;
				} else if (_feedItemType == "PulseStatus" && _inReplyToSeqNum != null){
					var displayName = $userDisplayService.getDisplayName(_feedItemData.inReplyToUserName, _feedItemData.inReplyToFirstName, _feedItemData.inReplyToLastName);
					commentedOn = $displayMessageService.translate("COMMENTED ON") + " " + displayName + " " + $displayMessageService.translate("STATUS");
				}

				return commentedOn;
			};

			this.getConversationID = function(){
				return _feedItemData.conversationID;
			};

			this.getCreateDate = function(){
				return _createDate;
			};

			this.getDisplayName = function() {

				return $userDisplayService.getDisplayName(_username, _firstName, _lastName)
			};

            this.getIconURL = function(){
                return _iconURL;
            };

			this.getLikeCount = function(){
				return _feedItemData.likeCount
			};

			this.getLocation = function(){
				return _feedItemData.userLocation;
			};

			this.getMessage = function(){
				return _feedItemData.body;
			};

			this.getPosition = function(){
				return _feedItemData.userTitle;
			};

			this.getProviderType = function(){
				return _feedItemData.feed;
			};

			this.getSequenceNumber = function(){
				return _feedItemData.seqNo;
			};

            this.getUsername = function() {
                return _username;
            };

			this.incrementCommentCount = function(){
				_feedItemData.commentCount += 1;
			};


			this.isFollowing = function() {
				return _feedItemData.isFollowing;
			};

			this.isLiked = function() {
				return _feedItemData.isLiked;
			};

			this.setIsFollowing = function(isFollowing) {
				_feedItemData.isFollowing = isFollowing;
			};

			this.setIsLiked = function(isLiked) {

				if (isLiked == true){
					_feedItemData.likeCount += 1;
				} else{
					_feedItemData.likeCount -= 1;
				}

				_feedItemData.isLiked = isLiked;
			};

            this.toString = function(){
                return _createDate.toLowerCase();
            };
        };

        return FeedItem;
    }]);
